Improvement/dtm guide (#473)
SoftwareAG/c8y-docs: content/dtm/localization.md

---
weight: 40
layout: bundle
title: Localization
---

> The **Localization** feature is provided in the DTM application. To use this feature, subscribe to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). It should then appear under [Subscribed applications](/users-guide/administration/#subscribed-applications) in your tenant.

Reviewer: Make the subscription a requirement admonition

Reviewer 2: Done

> Once you have enabled feature branding, open the **Localization** page on the left menu.

Reviewer: on the left menu > in the navigator

> To change the language for the **DTM** application click the **User settings** option under [User button](/users-guide/getting-started/#gui-features) and select the desired language in the drop down menu. You need to refresh your screen for the application to display the newly selected language.

Reviewer: Remove this description and add a link to the section where this is explained:

https://cumulocity.com/guides/users-guide/getting-started/#user-settings

> Additionally, if you want to add translations for the [Custom property](/dtm/asset-types/#property-library) or [Asset types](/dtm/asset-types/#asset-types) created, or if you want to add custom translations for the existing static text in the DTM application, same can be done using the **Localization** feature.

Reviewer: ```suggestion
If you want to add translations for the [custom properties](/dtm/asset-types/#property-library) or [asset types](/dtm/asset-types/#asset-types) created, or if you want to add custom translations for the existing static text in the DTM application, this can be done by using the Localization feature.
```

![Main view localization page](/images/dtm/localization/dtm-localozation-main-page.png)

> On your first time login, the **Localization** page will show a row named **Group** as key. If necessary, add a translation for "group" according to the explanation above.

Reviewer: ```suggestion
Initially, the **Localization** page shows a row called **Group** as key. If necessary, add a translation for "group" according to the explanation above.
```

Reviewer: according to the explanation above > which explanation, there has not been any so far

> Once you created all **custom properties** and **asset types**, then all they names are added as keys in the **Localization** page. You can further add the translations as required.

Reviewer: ```suggestion
When creating a custom property or asset type its name is added as a key in the **Localization** page. You can further add translations as required.
```

> In case you want to add new translation in the DTM application, same can be done by adding the text as key and adding the required translation for the respective languages.

Reviewer: ```suggestion
To add a new translation in the DTM application, you can add the text as key and add the translations as required.
```

##### Adding new Keys for translations

Add new keys and their respective translations using the **Add term** option on the top right.

![localization-add-term](/images/dtm/localization/dtm-localization-add-term.png)

Fill all the mandatory fields in the dialog window and click **Save**. The new key is saved including the added translations.

Next, click **Apply** to apply the changes.

> ##### Add / Edit translations

Reviewer: ```suggestion
##### Add/edit translations
```

 You can add or edit translations for all the keys in two ways:

* Hover over the respective column for the **Edit** icon to be visible. Click the icon to select the desired language and add or edit the translation.

* Click the **Edit icon** in each row to open the dialog window **Edit term translations**. Add or edit the translation for any language. Click **Save** to finish the process.

**Edit icon** is present for each row and on click “Edit term translations” screen loads.

Translation can be added or modified for any of the languages and the changes can be saved using **Save** button.

> Once all your translations are in place, change the language under **User settings**. The added translations will now be displayed in the DTM application.

Reviewer: Here we go with the link to the section where we explain how to change the language. Remove it from the top of this section.

{{< c8y-admon-info>}}

> The added translations for **Asset type** and **Custom property** will only show up in the **Assets** page.

Reviewer: This admonition is not closed

---
weight: 40
layout: bundle
title: Localization
---

> Localization feature is provided in **DTM (Digital Twin Manager)** (DTM) application. To use this feature, subscribe to [**Feature branding**](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). It should then appear under [subscribed applications](/users-guide/administration/#subscribed-applications) in your tenant.

Reviewer 2: ```suggestion
The **Localization** feature is provided in the DTM application. To use this feature, subscribe to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). It should then appear under [Subscribed applications](/users-guide/administration/#subscribed-applications) in your tenant.
```

> Once the Feature branding is enabled, navigate to **Localization** page using the left nav.

Reviewer 2: ```suggestion
Once you have enabled feature branding, open the **Localization** page on the left menu.
```

> Language for the **DTM** application can be changed by clicking on the **User settings** option under [User button](/users-guide/getting-started/#gui-features) and by choosing the language in the **Language** dropdown. Based on the chosen language, application loads accordingly.

Reviewer 2: ```suggestion
To change the language for the **DTM** application click the **User settings** option under [User button](/users-guide/getting-started/#gui-features) and select the desired language in the drop down menu. You need to refresh your screen for the application to display the newly selected language.
```

> Additionally, if you want to add translations for the [**Custom property**](/dtm/asset-types/#property-library) or [**Asset types**](/dtm/asset-types/#asset-types) created, or if you want to add custom translations for the existing Static text in **DTM** application, same can be done using Localization feature.

Reviewer 2: ```suggestion
Additionally, if you want to add translations for the [Custom property](/dtm/asset-types/#property-library) or [Asset types](/dtm/asset-types/#asset-types) created, or if you want to add custom translations for the existing static text in the DTM application, same can be done using the **Localization** feature.
```

> ![localization-mainpage](/images/dtm/localization/dtm-localozation-main-page.png)

Reviewer 2: ```suggestion
![Main view localization page](/images/dtm/localization/dtm-localozation-main-page.png)
```

> On first time login, when you navigate to **Localization** page, by default there will be a row added with **Group** as Key. You can add the translations if needed for **Group**.

Reviewer 2: ```suggestion
On your first time login, the **Localization** page will show a row named **Group** as key. If necessary, add a translation for "group" according to the explanation above.
```

> Once the Custom properties and asset types are created, then all those names are added as Keys in **Localization** page. You can further add the translations as required.

Reviewer 2: ```suggestion
Once you created all **custom properties** and **asset types**, then all they names are added as keys in the **Localization** page. You can further add the translations as required.
```

> In case you want to add new translation in the **DTM** application, same can be done by adding the text as Key and adding the required translation for the respective languages.

Reviewer 2: ```suggestion
In case you want to add new translation in the DTM application, same can be done by adding the text as key and adding the required translation for the respective languages.
```

##### **Adding new Keys for translations:**

> New keys and their respective translations can be added using the **Add term** option on top right.

Reviewer 2: ```suggestion
Add new keys and their respective translations using the **Add term** option on the top right.
```

> On click of **Add term** option, ADD TERM AND TRANSLATIONS screen loads.

Reviewer 2: delete

![localization-add-term](/images/dtm/localization/dtm-localization-add-term.png)

> Fill all the mandatory parameters and click on **Save** to save the changes. New key with added translations are saved.

Reviewer 2: ```suggestion
Fill all the mandatory fields in the dialog window and click **Save**. The new key is saved including the added translations.
```

> Next, click on **Apply** for the changes to be applicable.

Reviewer 2: ```suggestion
Next, click **Apply** to apply the changes.
```

##### **Add / Edit translations:**

>  For all the keys, you can add translations or edit the existing translations in two ways.

Reviewer 2: ```suggestion
 You can add or edit translations for all the keys in two ways:
```

> 1.	By clicking on the **Edit icon** against the column for the desired language:

Reviewer 2: ```suggestion
* Hover over the respective column for the **Edit** icon to be visible. Click the icon to select the desired language and add or edit the translation. 
```

> For every row, against each column **Edit icon** appears on hover.

Reviewer 2: delete

> On click of this **Edit icon**, you can either add new translation or edit existing translation text.

Reviewer 2: delete. combined with content above

> 2.	By clicking on the **Edit icon** against each row:

Reviewer 2: ```suggestion
* Click the **Edit icon** in each row to open the dialog window **Edit term translations**. Add or edit the translation for any language. Click **Save** to finish the process.
```

**Edit icon** is present for each row and on click “Edit term translations” screen loads.

Translation can be added or modified for any of the languages and the changes can be saved using **Save** button.

> Once all your translations are in place, change the language using the **Language** dropdown under **User settings**. The added translations will reflect in the **DTM** application.

Reviewer 2: ```suggestion
Once all your translations are in place, change the language under **User settings**. The added translations will now be displayed in the DTM application.
```

> >**Info:** The translations added for **Asset type** and **Custom property** will be reflecting only in **Assets** page.

Reviewer 2: ```suggestion
{{< c8y-admon-info>}}
The added translations for **Asset type** and **Custom property** will only show up in the **Assets** page.
```

{{< c8y-admon-info>}}

>  Currently the translations for DTM application are available for English and German language.

Reviewer 2: ```suggestion
The translations for the DTM application are currently available for English and German.
```

---
weight: 40
layout: bundle
title: Localization
---

{{< c8y-admon-related >}}
* [Subscribed applications](/users-guide/administration/#subscribed-applications) for more information on how to access the DTM application in your tenant.

{{< /c8y-admon-related >}}

<a name=""></a>
### Localization feature

The **Localization** feature is provided in the DTM application. To use this feature, subscribe to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). It should then appear under [Subscribed applications](/users-guide/administration/#subscribed-applications) in your tenant.

{{< c8y-admon-req >}}
* Subscribed to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview)
{{< /c8y-admon-req >}}

> To use the **Localization** feature in the DTM application, subscribe to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). Once you have enabled it, open the **Localization** page in the navigator.

Author: @MWindel 
To use the **Localization** feature in the DTM application, subscribe to [Feature branding](/users-guide/enterprise-tenant/#branding) in the [Enterprise tenant](/users-guide/enterprise-tenant/#overview). 
The above statement is repeated, may be we can delete it.

Reviewer 2: Done

Additionally, if you want to add translations for the [custom property](/dtm/asset-types/#property-library) or [asset types](/dtm/asset-types/#asset-types) created, or if you want to add custom translations for the existing static text in the DTM application, same can be done using the **Localization** feature.

![Main view localization page](/images/dtm/localization/dtm-localozation-main-page.png)

> Initially, the **Localization** page will show a row called **Group** as key. If necessary, add a translation for "group" according to the explanation above.

Author: @MWindel 
Change "group" to "Group" (with a capital G)

Reviewer 2: Done
